### PR TITLE
Treatment summary

### LINKED ISSUE #1212: Treatment summary
[interim treatment summary.pdf](https://github.com/coronasafe/care_fe/files/6546104/interim.treatment.summary.pdf)

An option to download a treatment summary must appear on the live consultation

The format must be strict as the attached doc. 

Most field there are self-explanatory. A few are explained below:
- Date of admission, must be the date of admission in that facility, i,e, in that consultation
- Date of positive must be the date the patient turned positive which could be before the admission
- Date of negativity may be left blank if the patient has not turned negative yet
- Comorbidities is captured in "Any medical history" in patient registration form
- Diagnosis, must include "History of present illness", "Examination details and Clinical conditions" and "Diagnosis" from the consultation form. It must also include "Physical Examination Info" from the consultation update. 
- Treatment summary includes 
                                 - table of "date and SPO2, temperature from consultation update
                                 - treatment summary from the consultation page. 

This interim treatment summary must preview on the system like the KASP referral letter in shifting and must be available to download as PDF

### PR DESCRIPTION
**Pull request by divyagar**
Closes #1212

Add treatment summary to live consultation. User can also download pdf of this summary.

https://user-images.githubusercontent.com/42246254/120127194-5d775680-c1dc-11eb-96cf-3f454df181ee.mp4

### CODE REVIEW

**Review by arpancodes — CHANGES_REQUESTED** [review]
Everything works fine! Just one thing, after we save or even cancel pdf download the button goes white, it is a focus issue, because it becomes green again after we click somewhere else.
treatment_summary

**Comment by aparnacoronasafe** [comment]
@bodhish Preview has failed. Please merge to staging and leave a "test in staging" label

**Comment by divyagar** [comment]
@arpancodes Thanks for the feedback.
@aparnacoronasafe @bodhish I have fixed the issue.

**Comment by bodhish** [comment]
@aparnacoronasafe what difference will it make if I merge it to staging 🤔

**Review by arpancodes — APPROVED** [review]
Looks good to me

File: src/Components/Facility/ConsultationDetails.tsx [hunk 1]
```diff
@@ -225,325 +227,349 @@ export const ConsultationDetails = (props: any) => {
...
+                            Dosage
+                          </th>
+                          <th className="px-6 py-3 border-b border-gray-200 bg-gray-50 text-left text-xs leading-4 font-medium text-gray-500 uppercase tracking-wider">
+                            Days
+                          </th>
+                        </tr>
+                      </thead>
+                      <tbody>
+                        {consultationData.discharge_advice.map((med: any) => (
+                          <tr className="bg-white">
```
Comment (smartclash) [inline src/Components/Facility/ConsultationDetails.tsx]: Add `key` to the table row.

File: src/Components/Facility/TreatmentSummary.tsx [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+                                                            <th className="border border-gray-800">
+                                                                Details
+                                                            </th>
+                                                        </tr>
+                                                    </thead>
+                                                    <tbody>
+                                            {patientData.medical_history && patientData.medical_history.length > 0 ? (
+                                                patientData.medical_history.map((obj: any) => {
+                                                    return (
+                                                        <tr>
```
Comment (smartclash) [inline src/Components/Facility/TreatmentSummary.tsx]: Please add `key` to this table row

File: src/Components/Facility/TreatmentSummary.tsx [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+                                                            </td>
+                                                        </tr>
+                                                    )
+                                                })) : (
+                                                <tr>
+                                                    <td className="border border-gray-800 text-center">---</td>
+                                                    <td className="border border-gray-800 text-center">---</td>
+                                                </tr>
+                                            )}
+                                                    </tbody>
```
Comment (smartclash) [inline src/Components/Facility/TreatmentSummary.tsx]: Please fix this spacing issue

File: src/Components/Facility/TreatmentSummary.tsx [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+                                                    <th className="border border-gray-800 text-center">values range</th>
+                                                    <th className="border border-gray-800 text-center">unit</th>
+                                                </tr>
+                                            </thead>
+
+                                        <tbody>
+                                            {investigations.length > 0 ? (
+                                                Object.values(investigations).map((value: any) => {
+                                                    return (
+                                                        <tr>
```
Comment (smartclash) [inline src/Components/Facility/TreatmentSummary.tsx]: Add `key` here too

File: src/Components/Facility/TreatmentSummary.tsx [hunk 5]
```diff
@@ -0,0 +1,327 @@
...
+                                            ) : (
+                                                    <tr>
+                                                        <td className="border border-gray-800 text-center">---</td>
+                                                        <td className="border border-gray-800 text-center">---</td>
+                                                        <td className="border border-gray-800 text-center">---</td>
+                                                        <td className="border border-gray-800 text-center">---</td>
+                                                        <td className="border border-gray-800 text-center">---</td>
+                                                        <td className="border border-gray-800 text-center">---</td>
+                                                    </tr>
+                                            )}
```
Comment (smartclash) [inline src/Components/Facility/TreatmentSummary.tsx]: Fix spacing issue

File: src/Components/Facility/TreatmentSummary.tsx [hunk 6]
```diff
@@ -0,0 +1,327 @@
...
+                                                            Temperature
+                                                        </th>
+                                                    </tr>
+                                                </thead>
+
+                                                <tbody>
+                                            {
+                                                dailyRounds.length > 0 ? (
+                                                    dailyRounds.map((rounds: DailyRoundsModel) => (
+                                                        <tr>
```
Comment (smartclash) [inline src/Components/Facility/TreatmentSummary.tsx]: `key` here too

File: src/Components/Facility/TreatmentSummary.tsx [hunk 7]
```diff
@@ -0,0 +1,327 @@
...
+                                                ) : (
+                                                    <tr>
+                                                        <td className="border border-gray-800 text-center">---</td>
+                                                        <td className="border border-gray-800 text-center">---</td>
+                                                        <td className="border border-gray-800 text-center">---</td>
+                                                    </tr>
+
+                                                )
+                                            }
+                                                </tbody>
```
Comment (smartclash) [inline src/Components/Facility/TreatmentSummary.tsx]: Spacing issue, please fix :)

**Review by smartclash — CHANGES_REQUESTED** [review]

File: src/Components/Facility/TreatmentSummary.tsx (hunk 7 above)
Comment (divyagar) [reply src/Components/Facility/TreatmentSummary.tsx]: done

File: src/Components/Facility/ConsultationDetails.tsx (hunk 1 above)
Comment (divyagar) [reply src/Components/Facility/ConsultationDetails.tsx]: done

File: src/Components/Facility/TreatmentSummary.tsx (hunk 2 above)
Comment (divyagar) [reply src/Components/Facility/TreatmentSummary.tsx]: done

File: src/Components/Facility/TreatmentSummary.tsx (hunk 3 above)
Comment (divyagar) [reply src/Components/Facility/TreatmentSummary.tsx]: done

File: src/Components/Facility/TreatmentSummary.tsx (hunk 5 above)
Comment (divyagar) [reply src/Components/Facility/TreatmentSummary.tsx]: done

File: src/Components/Facility/TreatmentSummary.tsx (hunk 6 above)
Comment (divyagar) [reply src/Components/Facility/TreatmentSummary.tsx]: done

File: src/Components/Facility/TreatmentSummary.tsx (hunk 4 above)
Comment (divyagar) [reply src/Components/Facility/TreatmentSummary.tsx]: done

**Comment by divyagar** [comment]
@smartclash @bodhish Done with all changes.

**Comment by aparnacoronasafe** [comment]
saved Relevant investigations does not appear on the treatment summary

**Comment by amruthavijay88** [comment]
Tested. 'Examination details and Clinical conditions' were missing in the print out version - though it was updated.

**Comment by divyagar** [comment]
@amruthavijay88 @aparnacoronasafe I have made changes.

**Pull request merged**